include/aspect/velocity_boundary_conditions/ascii_data.h

```diff
@@ -0,0 +1,519 @@
+/*
+  Copyright (C) 2011 - 2014 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__velocity_boundary_conditions_ascii_data_h
+#define __aspect__velocity_boundary_conditions_ascii_data_h
+
+#include <aspect/velocity_boundary_conditions/interface.h>
+#include <aspect/simulator_access.h>
+
+#include <aspect/geometry_model/box.h>
+#include <aspect/geometry_model/spherical_shell.h>
+#include <aspect/utilities.h>
+
+#include <deal.II/base/table.h>
+#include <deal.II/base/table_indices.h>
+#include <deal.II/base/function_lib.h>
+
+#include <fstream>
+
+
+namespace aspect
+{
+  namespace VelocityBoundaryConditions
+  {
+    using namespace dealii;
+
+    namespace internal
+     {
+       namespace
+       {
+         bool
+         fexists(const std::string &filename)
+         {
+           std::ifstream ifile(filename.c_str());
+           return !(!ifile); // only in c++11 you can convert to bool directly
```

Comment:
It doesn't matter, but you can always do `static_cast<bool>(ifile)`. C++11 allows you to do the conversion implicitly while C++98 requires you to make it explicit here.